pkg/controllers/k8sdefaultendpoint/syncer.go

```diff
@@ -0,0 +1,111 @@
+package k8sdefaultendpoint
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/loft-sh/vcluster/pkg/util/loghelper"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type K8SDefaultEndpointReconciler struct {
+	LocalClient   client.Client
+	VirtualClient client.Client
+	Log           loghelper.Logger
+}
+
+func (e *K8SDefaultEndpointReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	err := syncKubernetesServiceEndpoints(ctx, e.VirtualClient, e.LocalClient)
+	if err != nil {
+		return ctrl.Result{RequeueAfter: time.Second}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager adds the controller to the manager
+func (e *K8SDefaultEndpointReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("kubernetes_default_endpoint").
+		For(&corev1.Endpoints{}).
+		Watches(&source.Kind{
+			Type: &corev1.Endpoints{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      "kubernetes",
+					Namespace: "default",
+				},
```

Comment:
This does not ensure that reconcile is only triggered for the `default/kubernetes`.
You need something like this:
https://github.com/loft-sh/vcluster/blob/7c5c470e87dc8224f41161b98f4a8ecee0d27ea4/pkg/controllers/coredns/nodehosts.go#L86-L90
+
https://github.com/loft-sh/vcluster/blob/7c5c470e87dc8224f41161b98f4a8ecee0d27ea4/pkg/controllers/coredns/nodehosts.go#L101
And then `.Watches()` can be removed.

Reply:
Also, because this controller is being registered with the host side manager (`controller.SetupWithManager(ctx.LocalManager)`) - it should care about Endpoint resource that is named the same as vcluster service, and is in the vcluster namespace, not the `default/kubernetes` one.